packages/eslint-plugin/src/util/explicitReturnTypeUtils.ts

```diff
@@ -194,11 +216,8 @@ function isTypedFunctionExpression(
   }
 
   return (
-    isTypeAssertion(parent) ||
-    isVariableDeclaratorWithTypeAnnotation(parent) ||
-    isPropertyDefinitionWithTypeAnnotation(parent) ||
+    isTypedParent(parent, node) ||
     isPropertyOfObjectWithType(parent) ||
-    isFunctionArgument(parent, node) ||
     isConstructorArgument(parent)
```

Comment:
DRYing these out, it makes it more obvious that the recursive object-property doesn't include this `isConstructorArgument()` case, but it seems like it should?

Happy to move it into `isTypedParent()` if this is indeed a bug.